pkg/controller/kubebuilderx/controller.go

```diff
@@ -88,36 +88,43 @@ func (c *controller) Do(reconcilers ...Reconciler) Controller {
 			return c
 		}
 
-		c.tree, c.err = reconciler.Reconcile(c.tree)
-		if c.err != nil {
+		c.res, c.err = reconciler.Reconcile(c.tree)
+		if c.err != nil || c.res.Next == cmmt || c.res.Next == rtry {
 			return c
 		}
 	}
 
 	return c
 }
 
-func (c *controller) Commit() error {
+func (c *controller) Commit() (ctrl.Result, error) {
 	defer c.emitFailureEvent()
 
-	if c.err != nil && !intctrlutil.IsDelayedRequeueError(c.err) {
-		return c.err
+	if c.err != nil {
+		return ctrl.Result{}, c.err
 	}
 	if c.oldTree.GetRoot() == nil {
-		return nil
+		return ctrl.Result{}, nil
 	}
 	builder := NewPlanBuilder(c.ctx, c.cli, c.oldTree, c.tree, c.recorder, c.logger)
-	if err := builder.Init(); err != nil {
-		return err
+	if c.err = builder.Init(); c.err != nil {
+		return ctrl.Result{}, c.err
 	}
-	plan, err := builder.Build()
-	if err != nil {
-		return err
+	var plan graph.Plan
+	plan, c.err = builder.Build()
+	if c.err != nil {
+		return ctrl.Result{}, c.err
+	}
+	if c.err = plan.Execute(); c.err != nil {
+		if apierrors.IsConflict(c.err) {
```

Comment:
There could be other types of ephemeral errors that may retry later, e.g. IsTimeout, IsServerTimeout, IsTooManyRequests

Reply:
Here we translate the optimistic locking conflict error to a retry-immediately event, in aim to eliminate the annoying message in the log.
All errors by default will trigger a exponential based retry operation.